File: generator/src/main/java/ch/bailu/gtk/tag/TypeTag.java
package ch.bailu.gtk.tag;

public class TypeTag extends NamedTag {
    private String type;

    public TypeTag(Tag parent) {
        super(parent);
    }


    @Override
    public void setAttribute(String name, String value) {
        if ("type".equals(name)) {
            type = value;
        } else {
            super.setAttribute(name, value);
        }

    }


    public String getType() {
        return type;
    }
}

File: generator/src/main/kotlin/ch/bailu/gtk/parser/tag/TypeTag.kt
package ch.bailu.gtk.parser.tag

open class TypeTag(parent: TagWithParent) : NamedWithDocTag(parent) {
    var type: String = ""
        private set

    override fun setAttribute(name: String, value: String) {
        if ("c:type" == name) {
            type = value
        } else {
            super.setAttribute(name, value)
        }
    }
}
